File: JAVA/SubSequenceLogic.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Map;


public class SubSequenceLogic {
	public static void main(String[] args) {
		String[] splited = "abcde".split("");
		
		List<String> binarys = new ArrayList<>();
		List<String> toReturn = new ArrayList<String>();
		
		String binaryResult = "";
		for(int i = 0; i < splited.length; i++){
			binaryResult += "1";
		}
		
		Integer init = 0;
		
		do{
			binarys.add(String.format("%0" + splited.length + "d", Integer.valueOf(Integer.toBinaryString(init))));
			init++;
		}while(!String.format("%0" + splited.length + "d", Integer.valueOf(Integer.toBinaryString(init))).equals(binaryResult) );
		
		binarys.add(binaryResult);
		
		for(String s : binarys){
			String[] bSplited = s.split("");
			String toConcat = "";
			
			for(int i = 0; i < bSplited.length; i++){
				if(bSplited[i].equals("1")){
					toConcat += splited[i];
				}
			}
			if(toConcat.length() > 0){
				toReturn.add(toConcat);
			}
		}
		
		Collections.sort(toReturn);

		for (String sss : toReturn) {
			System.out.println(sss);
		}
	}
}

File: JS/SubSequenceLogic.js
var splitted = "abcde".split("");

var binarys = [];
var toReturn = [];

var binaryResult = "";
var pad = "";

for(var i =0; i < splitted.length; i++){
	binaryResult += 1;
	pad += "0";
}

var init = 0;
var lastResult = "";

do{
	var bin = init.toString(2);
	var result = (pad + bin).slice(-pad.length);
	lastResult = result;
	binarys.push(result);
	init++;
}while(lastResult != binaryResult);

binarys.forEach(function(s){
	var bSplited = s.split("");
	var toConcat = "";
	for(var i = 0; i < bSplited.length; i++){
		if(bSplited[i] == "1"){
			toConcat += splitted[i];
		}
	}
	if(toConcat.length > 0){
		toReturn.push(toConcat);
	}
})

console.log(toReturn.sort());

